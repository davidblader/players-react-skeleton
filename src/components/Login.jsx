import React from 'react';

class Login extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      loginFields: {
        email: '',
        password: '',
      },
    };
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  handleSubmit(e) {
    e.preventDefault();
    const { email, password } = this.state.loginFields;
    console.log(email);
    console.log(password);
  }

  handleChange(e) {
    const { loginFields } = this.state;
    loginFields[e.target.name] = e.target.value;
    this.setState({ loginFields });
  }

  render() {
    return (
      <div id="login">
        <h2>Login</h2>
        <div id="login-inputs">
          <form onSubmit={this.handleSubmit}>
            <div><input placeholder="Email" name="email" type="text" onChange={this.handleChange} /></div>
            <div><input placeholder="Password" name="password" type="password" onChange={this.handleChange} /></div>
            <div><input type="submit" value="Login" /></div>
          </form>
        </div>
      </div>
    );
  }
}

export default Login;
