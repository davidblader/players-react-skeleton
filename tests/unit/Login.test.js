import React from 'react';
import renderer from 'react-test-renderer';
import Login from '../../src/components/Login';

test('Login page loads and matches snapshot', () => {
  const loginComponent = renderer.create(<Login />);
  const tree = loginComponent.toJSON();
  expect(tree).toMatchSnapshot();
});

