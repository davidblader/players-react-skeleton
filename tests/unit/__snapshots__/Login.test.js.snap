// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page loads and matches snapshot 1`] = `
<div
  className="animal-crossing-box"
>
  <span
    className="animal-crossing-header"
  >
    Login
  </span>
  <div
    id="login-inputs"
  >
    <form
      onSubmit={[Function]}
    >
      <div>
        <span
          className="hide"
        >
          Email
        </span>
        <input
          id="email"
          name="email"
          onChange={[Function]}
          placeholder="Email"
          required={true}
          type="text"
        />
      </div>
      <div>
        <span
          className="hide"
        >
          Password
        </span>
        <input
          id="password"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
        />
      </div>
      <div
        className="btn-container"
      >
        <button
          className="animal-crossing-btn"
          id="login"
          type="submit"
        >
          Login
        </button>
      </div>
    </form>
  </div>
</div>
`;
